### DISCUSSION TITLE
Allow using compressed small font

### DISCUSSION THREAD

**Discussion by danyallard** [ni/labview-icon-editor#224] [General]
As a LabVIEW user I want to be able to enter more characters in 32 pixel icons.

Small font usually use 1-4 pixel per character. It's possible to compress some of the characters to use 2 instead of 3 pixel. 3 instead of 4. 

E.g. for L

![image](https://github.com/ni/labview-icon-editor/assets/3820056/d77097fc-8c6a-4a8c-9849-772cea480fe4)

![image](https://github.com/ni/labview-icon-editor/assets/3820056/afcc8763-82fe-44e8-a918-f10610627347)

E.g. for E

![image](https://github.com/ni/labview-icon-editor/assets/3820056/56b93cca-164d-454f-8fcf-243c92603fb1)

![image](https://github.com/ni/labview-icon-editor/assets/3820056/d297a0eb-1cbf-4cd5-b964-c9f5be0189eb)

**Comment by mballawickersigpi** [comment]
@danyallard I also want to add this function in the IE. I think I remember seeing posts in the past where you have taken the current IE code and updated it with features. Have you implemented this function in your updates? Any incites you can share?

How do you prioritize which letter/letters to compress?
How do you determine how wide or thin a letter can be?
Do you want to include lowercase letters?
Do you only want to use this and discard all other fonts?

For my custom Icon editor I had to create my own set of bit maps for each letter (Wide, Med and Thin) and associate a priority number with each letter.

![Image](https://github.com/user-attachments/assets/95b53aba-593f-4b54-b5da-8572969fea67)

My sizing vi sets all characters to thin and if there are extra spaces it increase the font size of the highest priority letter and rechecks the size.

![Image](https://github.com/user-attachments/assets/6da2ecb3-dc0d-453c-8c28-a9e8472717db)

I was thinking to add an "IE Adaptable"  font to the ni IE font menu and when selected would compress the letters to fit the space width.

Additional input would be welcome.

**Comment by svelderrainruiz** [comment]
We marked this as issue needs more information, but did not mentioned specifically which information is needed. I am adding this to our agenda again.

**Comment by jimkring** [comment]
Somewhat related... I have drafted a PR to add Small Fonts support for Mac and Linux via a Pixel Font: https://github.com/ni/labview-icon-editor/pull/433.  This could be the beginnings of a more complete set of Pixel Fonts that includes a "Small Fonts Compressed".  To do this (beyond or extending PR I created) we would need to add User Interface features to the Icon Editor for selecting "Compressed" (e.g. perhaps beneath the "Capitalize" option) as some more refactoring and organization of the underlying Pixel Font API that I cobbled together.
